### PR TITLE
Add more info to readme about running individual programs in the repo

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,20 +45,69 @@ You can either keep running the code from source, using `cargo run --release --b
...
 For normal operation (replace testnet by mainnet to run the mainnet node)
+You don't need to specify RUST_LOG environment variable for logging, since it's by default set to be `INFO`.
+
 Note: spaces don't matter, so these are aligned for readability
-- Node daemon: `RUST_LOG=info cargo run --release --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log`
-- CLI Wallet:  `RUST_LOG=info cargo run --release --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log`
-- RPC Wallet:  `RUST_LOG=info cargo run --release --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log`
-- GUI:         `RUST_LOG=info cargo run --release --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log`
+
+- Node daemon:
```
Comment (azarovh) [inline README.md:75]: I remember people were confused that you need to run either daemon + wallet or just gui. Maybe worth mentioning somewhere.

**Review by azarovh — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -45,20 +45,69 @@ You can either keep running the code from source, using `cargo run --release --b
 
 The logging of mintlayer-core is configured via the `RUST_LOG` environment variable. All log messages are printed to the terminal screen; we prefer simplicity over complicated log machinery. For example, to see all logs of the `info` level and above (the default level for normal operation), you can run the node with `RUST_LOG=info cargo run --bin node-daemon -- testnet`. If you're facing an issue, it's recommended to use `RUST_LOG=debug` instead. We recommend using these commands that not only print the logs on the screen, but also write them to a file in case you face an issue. On Linux, this can be achieved using `tee` as shown below.
 
+If the `RUST_LOG` environment variable is not specified, the debug level `info` will be used by default.
```
[the remark this answers is not in the document]
Comment (TheQuantumPhysicist) [reply README.md:48]: Done. Thanks.

File: README.md (hunk 1 above)
Comment (TheQuantumPhysicist) [reply README.md:75]: Fair. Will do. Thanks.

File: README.md [hunk 3]
```diff
@@ -31,6 +31,17 @@ You can see the full list of mintlayer executables by running `cargo run --bin`,
...
+### What do you need to run?
+
+There are many ways you can manage running the components of mintlayer-core. The following are a few examples.
+
+You can:
+
+- Run the node-daemon + wallet-cli to manage your coins, where the wallet-cli owns your wallet file and manages your coins. The wallet-cli itself can optionally act as an RPC server
+- Run the node-daemon + wallet-rpc-daemon + wallet-cli, where both daemons will be services in your system, the wallet-rpc-daemon will own your wallet (and start staking automatically for you), and the wallet-cli will just be an easy interface for communicating with your wallet
+- Run only the node-gui, where the GUI contains both the node and the wallet. This is a good option for non-techy users
+- Run the node-daemon with the API server components ([see this readme](api-server/README.md)) without a wallet, to manage your application
```
Comment (ImplOfAnImpl) [inline README.md:43]: A nitpick: I wouldn't use `the` with the names of applications, because they are basically proper names (I mean technically a proper name can be used with `the` in English, like `The Eiffel Tower`, but an application name is more like a person's name to me and `the` is not used with those). Alternatively, I'd remove the hyphen to make the name a common name. So, I'd write

> Run the node daemon + wallet cli to manage your coins, where the wallet cli owns your wallet file and manages your coins.

or

> Run node-daemon + wallet-cli to manage your coins, where wallet-cli owns your wallet file and manages your coins.

Comment (ImplOfAnImpl) [reply README.md:43]: Also, there are no full stops at the end of the sentences.

File: README.md [hunk 4]
```diff
@@ -45,20 +56,69 @@ You can either keep running the code from source, using `cargo run --release --b
...
+```sh
+git checkout tags/v0.3.0
+```
+
+Release branches are more recommended than tags, because they get necessary security patches, if any. Tags are just markers.
+
+After having checked out the version you need, the following is how to run the software.
+
 For normal operation (replace testnet by mainnet to run the mainnet node)
+You don't need to specify RUST_LOG environment variable for logging, since it's by default set to be `INFO`.
```
Comment (ImplOfAnImpl) [inline README.md:82]: The line `For normal operation (replace testnet by mainnet to run the mainnet node)` looks off, perhaps it should be made a header

Also a nitpick: IMO `replace with` would sound better than `replace by` in this context.

File: README.md [hunk 5]
```diff
@@ -45,20 +56,69 @@ You can either keep running the code from source, using `cargo run --release --b
...
 Note: spaces don't matter, so these are aligned for readability
-- Node daemon: `RUST_LOG=info cargo run --release --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log`
-- CLI Wallet:  `RUST_LOG=info cargo run --release --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log`
-- RPC Wallet:  `RUST_LOG=info cargo run --release --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log`
-- GUI:         `RUST_LOG=info cargo run --release --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log`
+
+- Node daemon:
+```sh
+cargo run --release --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log
+```
```
Comment (ImplOfAnImpl) [inline README.md:89]: Here and below, the examples that belong a list item should be indented by a tab, otherwise they don't look like they belong to the list (at least not in VSCode's preview for markdown)

File: README.md [hunk 6]
```diff
@@ -45,20 +56,69 @@ You can either keep running the code from source, using `cargo run --release --b
...
+- RPC Wallet:
+```sh
+RUST_LOG=debug cargo run --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log
+```
+- GUI:
+```sh
+RUST_LOG=debug cargo run --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log
+```
+
+There are more complex logging filtering that can be done at the crate level, but this is out of scope.
```
Comment (ImplOfAnImpl) [inline README.md:122]: `are` and `filtering` don't add up. I'd rephrase this, e.g. as follows:

> More complex log filtering is possible, e.g. you can filter out log lines produced by a certain crate or module, but this is out of scope of this document.

**Review by ImplOfAnImpl — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (TheQuantumPhysicist) [reply README.md:122]: Done. Thanks.

File: README.md (hunk 3 above)
Comment (TheQuantumPhysicist) [reply README.md:43]: Fixed. The full stops though is intended because it's a list. AFAIK, lists don't follow these rules. Not sure to be honest.

File: README.md (hunk 5 above)
Comment (ImplOfAnImpl) [reply README.md:89]: Well, I've created a temporary branch to check how GitHub renders it and there is a difference.
E,g, your branch - https://github.com/mintlayer/mintlayer-core/tree/fix/readme-run?tab=readme-ov-file#assuming-youre-using-the-source-code
My branch - https://github.com/mintlayer/mintlayer-core/tree/tmp-readme-run-check-doc-rendering#assuming-youre-using-the-source-code

The difference isn't huge but IMO it's still better this way.

Comment (TheQuantumPhysicist) [reply README.md:89]: Alrighty. Will do it.
